oak_functions/loader/src/server.rs

```diff
@@ -905,22 +969,20 @@ mod tests {
 
     #[tokio::test]
     async fn test_hosted_channel_read_no_message() {
-        let channel_handle = ChannelHandle::LookupData as i32;
+        let channel_handle = ChannelHandle::Testing as i32;
```

Comment:
Why does it matter what handle is used here? Is the test wasm state expected to write to that channel (it didn't seem the case to me, maybe I missed it)?

Reply:
Because I changed the keys in `extension_endpoints` in `WasmState` to `ChannelHandle` (instead of the `ABI_FUNCTION_NAME`, which UwabiExtensions don't have). 

In the next PR I want to give the endpoints to the extensions themselves, then this will change.

Reply:
I think it is better not to have test-specific variants in the proto spec. I'd suggest adding a TODO to the proto about removing `TESTING`.

Reply:
I see your point of not having a test handle in the proto definition. 

But, I would love to have a testing extension to be able to use it in integration tests with the Wasm module for e.g., [here](https://github.com/project-oak/oak/blob/main/oak_functions/sdk/oak_functions/tests/integration_test.rs).

And I see no way except for providing a Test Handle in the proto. 

I think it is justifiable, as we will have some handles which are not used in every application (e.g., Metric/TF).

Can you think of any other way?

Reply:
I think it would be very convenient to have an extension for testing (you could think of it as a debug extension, too). 

The way I imagine this to work is that the Wasm module can send an aribitrary string, e.g., "Refresh the Lookup Data" to trigger some behaviour we need for testing and which we implement in the testing extension.

I believe, if we want to use that testing extension from a Wasm Module, there is no way without adding the Testing (or Debug) ChannelHandle to the proto. 

I suggest I try to implement what I have in mind with the Testing Extension for now and we re-evaluate (also you are right, it is nice to have it anyway as a Uwabi Extension)

Reply:
Sounds good to me if the idea is to allow WASM modules to use it too, but I think we have to make sure that it is only available in the `unsafe` version.